core/src/main/java/com/scalar/db/transaction/consensuscommit/TransactionalTableMetadataManager.java

```diff
@@ -0,0 +1,87 @@
+package com.scalar.db.transaction.consensuscommit;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.scalar.db.api.DistributedStorageAdmin;
+import com.scalar.db.api.Operation;
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.exception.storage.ExecutionException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+
+public class TransactionalTableMetadataManager {
+
+  private final LoadingCache<TableKey, Optional<TransactionalTableMetadata>> tableMetadataCache;
+
+  public TransactionalTableMetadataManager(
+      DistributedStorageAdmin admin, long cacheExpirationTimeSecs) {
+
+    CacheBuilder<Object, Object> builder = CacheBuilder.newBuilder();
+    if (cacheExpirationTimeSecs > 0) {
+      builder.expireAfterWrite(cacheExpirationTimeSecs, TimeUnit.SECONDS);
+    }
+    tableMetadataCache =
+        builder.build(
+            new CacheLoader<TableKey, Optional<TransactionalTableMetadata>>() {
+              @Override
+              public Optional<TransactionalTableMetadata> load(@Nonnull TableKey key)
+                  throws ExecutionException {
+                TableMetadata tableMetadata = admin.getTableMetadata(key.namespace, key.table);
+                if (tableMetadata == null) {
+                  return Optional.empty();
+                }
+                return Optional.of(new TransactionalTableMetadata(tableMetadata));
+              }
+            });
+  }
+
+  /**
+   * Returns a transactional table metadata corresponding to the specified operation.
+   *
+   * @param operation an operation
+   * @return a table metadata. null if the table is not found.
+   * @throws ExecutionException if the operation failed
+   */
+  public TransactionalTableMetadata getTransactionalTableMetadata(Operation operation)
```

Comment:
Isn't it more general to make it `getTransactionalTableMetadata(String namespace, String table)`?

Reply:
Actually, we already have `TableMetadataManager` similar to this class, and it also has `getTransactionalTableMetadata(Operation operation)`:
https://github.com/scalar-labs/scalardb/blob/554e408984380131ca8d270bb95352515fb5b077/core/src/main/java/com/scalar/db/util/TableMetadataManager.java#L42

For consistency, I will keep the method. And we can change it in the future if needed. Thank you!